packages/vuetify/src/components/VDataTable/VDataTable.ts

```diff
@@ -241,7 +243,9 @@ export default mixins(
           },
         },
         on: {
-          // TODO: for click, the first argument should be the event, and the second argument should be data,
+          ...this.getDefaultMouseEventHandlers(':row', () => data),
+          // TODO: for click, contextmenu and dblclick, the first argument should be data,
+          // and the second argument should be the event,
```

Comment:
I didn't realise calendar was like this too. This is actually backwards, the first argument has to be the event so modifiers like `.prevent` work.